diff --git a/chrome/browser/content_settings/generated_notification_pref_unittest.cc b/chrome/browser/content_settings/generated_notification_pref_unittest.cc
index 678fac52cd9bec1d88b799a16a549df5f51c2a61..01774a71cb3a27981675df9c2195db030eb667b6 100644
--- a/chrome/browser/content_settings/generated_notification_pref_unittest.cc
+++ b/chrome/browser/content_settings/generated_notification_pref_unittest.cc
@@ -188,6 +188,7 @@ void SetupManagedTestConditions(
     case content_settings::SETTING_SOURCE_SUPERVISED:
       provider_type = HostContentSettingsMap::SUPERVISED_PROVIDER;
       break;
+    BRAVE_SETTING_SOURCE_REMOTE_LIST
     case content_settings::SETTING_SOURCE_USER:
     case content_settings::SETTING_SOURCE_NONE:
     case content_settings::SETTING_SOURCE_ALLOWLIST:
