diff --git a/components/content_settings/core/browser/host_content_settings_map.cc b/components/content_settings/core/browser/host_content_settings_map.cc
index 5d066ff91e41f83e6b3cff8b76c2021f65e690c4..d6649bcd727c01107bc89f1eee7008d5c9384d2e 100644
--- a/components/content_settings/core/browser/host_content_settings_map.cc
+++ b/components/content_settings/core/browser/host_content_settings_map.cc
@@ -305,6 +305,8 @@ HostContentSettingsMap::HostContentSettingsMap(PrefService* prefs,
   content_settings_providers_[ProviderType::kDefaultProvider] =
       std::move(default_provider);
 
+  BRAVE_CREATE_REMOTE_LIST_PROVIDER
+
   MigrateSettingsPrecedingPermissionDelegationActivation();
   if (should_record_metrics) {
     RecordExceptionMetrics();
